scripts/fetch/map_data.R

```diff
@@ -0,0 +1,42 @@
+fetchTimestamp.map_data_states <- fetchTimestamp.map_data_counties <- vizlab::alwaysCurrent
+
+#' Gets data for state polygons.
+#' @param viz a vizlab object including a \code{spatial_metadata} parameter input
+#' @details 
+#' Depends on: \code{spatial_metadata} which must include:
+#' /describe{
+#'   \item{bbox}{numeric in xmin, ymin, xmax, ymax order}
+#'}
+#' and optionally: 
+#' /describe{
+#'   \item{crs}{valid crs for \pkg{sf}}
+#' }
+fetch.map_data_states <- function(viz = as.viz("map_data_states")){
+
+  sm <- spatial_meta(viz)
+  
+  saveRDS(get_map_data_sf(database = 'state', 
+                          crs = sm$crs, 
+                          within = sm["bbox_polygon"]), 
+          viz[['location']])
+  }
+
+#' Gets data for county polygons.
+fetch.map_data_counties <- function(viz = as.viz("map_data_counties")){
```

Comment:
Lots of duplicated code here, and I assume the same would be true for when islands are implemented w/ this pattern. 

Depending on how we feel about the viz.yaml and adding params, personally I would rather see this as one function `fetch.map_data` that has map args (such as `database = state`)

Reply:
I hadn't thought of doing that... makes more sense.